### PR TITLE
fix(datasets): give possibility to add dataset with slashes in name

### PR DESCRIPTION
**Pull request by Always-prog**
### SUMMARY
Give possibility to users to add datasets which contains slashes in a table name

#### Before

https://github.com/apache/superset/assets/66589759/dcf616e9-6775-4e88-8d7e-22d7e06cb00b

#### After

https://github.com/apache/superset/assets/66589759/2e850696-45ab-40e8-b74e-ede9726ab685

### TESTING INSTRUCTIONS
1. Go to "*/dataset/add/" URL, or push "+DATASET" button.
2. Choose database and schema.
3. Choose a table with "/" symbol in name

### ADDITIONAL INFORMATION

- [x] Has associated issue: [23540](https://github.com/apache/superset/issues/23540)
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by michael-s-molina** [comment]
Thank for the PR @Always-prog. Could you add a test?

**Comment by michael-s-molina** [comment]
@Always-prog If you do a quick search for `<table_name>` you'll see that there are other endpoints that need to be modified.

@dpgaspar Is there any concern with this change?

**Comment by Always-prog** [comment]
@michael-s-molina I added a test for the slash in the table name and updated the table_name parameter type to `path` in all endpoints.

**Comment by michael-s-molina** [comment]
Here's the list of places that have `<table_name>`:

```
superset/databases/api.py::table_metadata
superset/databases/api.py::table_extra_metadata
superset/databases/api.py::select_star
superset/views/core.py::fetch_datasource_metadata
superset/db_engine_specs/presto.py::TABLE_DOES_NOT_EXIST_REGEX
```

**Comment by Always-prog** [comment]
@michael-s-molina I added `path` type to the all endpoints in your list excluding presto.py file because it contains just regex functions with <table_name>, it isn't endpoint.

**Comment by Always-prog** [comment]
@michael-s-molina Hello! Can I get code review?

File: tests/integration_tests/databases/api_tests.py [hunk 1]
```diff
@@ -686,6 +686,21 @@ def test_if_ssh_tunneling_flag_is_not_active_it_raises_new_exception(
         # the DB should not be created
         assert model is None
 
+    def test_get_table_details_with_slash_in_name(self):
+        table_name = "table_with/slash"
+        database = get_example_database()
+        if database.backend in ("mysql", "sqlite"):
```
Comment (michael-s-molina) [inline tests/integration_tests/databases/api_tests.py:692]: It seems [MySQL](https://stackoverflow.com/questions/9801760/using-slash-in-mysql-database) and [SQLite](https://stackoverflow.com/questions/3694276/what-are-valid-table-names-in-sqlite) support table names with slashes but you need to escape them.

**Comment by Always-prog** [comment]
@michael-s-molina I added MySQL and SQLite to the test case, but why is there an error while migrating? I haven't changed any migrations.

**Comment by michael-s-molina** [comment]
> @michael-s-molina I added MySQL and SQLite to the test case, but why is there an error while migrating? I haven't changed any migrations.

It seems the error was fixed by https://github.com/apache/superset/pull/24832. You need to rebase your PR.

**Comment by Always-prog** [comment]
@michael-s-molina Rebased

**Review by michael-s-molina — APPROVED** [review]
LGTM. Thank you for the fix @Always-prog!

**Pull request merged**